### ISSUE TITLE
Error: UPGRADE FAILED: failed to create resource: Deployment.apps "windmill-workers-dedicated-mydedicatedWorker" is invalid:

### ISSUE THREAD

**Issue by GrbrELO** [windmill-labs/windmill-helm-charts#37]
Sorry to bother you again, but the next error appeared. The error message is `Error: UPGRADE FAILED: failed to create resource: Deployment.apps "windmill-workers-dedicated-mydedicatedWorker" is invalid: metadata.name: Invalid value: "windmill-workers-dedicated-mydedicatedWorker": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')`

In short, here your default value is written in camelCase https://github.com/windmill-labs/windmill-helm-charts/blob/main/charts/windmill/values.yaml#L107 which is a bad idea, when that ends up to be part of name of an k8s resource respectively deployment. 

See
https://github.com/windmill-labs/windmill-helm-charts/blob/main/charts/windmill/templates/dedicated-workers.yaml#L7C43-L7C43

No need to hurry, I can work around that by setting a value by myself but I think it is better to fix it for the future.

**Comment by rubenfiszel** [comment]
I also added a validation linter so this shouldn't reproduce in the future. Thanks for the report
